torch/_subclasses/fake_tensor.py

```diff
@@ -320,6 +322,79 @@ def mk_fake_tensor(make_meta_t):
         )
         if out is NotImplemented:
             raise UnsupportedFakeTensorException("meta converter nyi")
+
+        from torch._dynamo.source import RandomValueSource
+
+        # TODO: extend this for smaller dtypes too.  Some care will have to be
+        # taken though, because smaller dtypes will compute differently but
+        # Python sympy compute is always done at arbitrary precision.
+        value = None
+        if (
+            not self.export
+            and type(t) is torch.Tensor
+            # TODO: It sure would be nice to test if something was a plain
+            # tensor without having to enumerate all the cases
+            and t.layout == torch.strided
+            and not (
+                t.is_sparse
+                or t.is_nested
+                or is_functorch_wrapped_tensor(t)
+                or is_legacy_batchedtensor(t)
+                or torch._is_functional_tensor(t)
+            )
+            and t.dim() == 0
+            and t.device.type == "cpu"
+            and t.dtype in [torch.int64, torch.float64]
+            and source is not None
+            # Impede setting up item() on things coming from random.  These
+            # are not "real" item() calls, instead UnspecializedPythonVariable
+            # is unsafely pretending an int is a tensor, which can sometimes
+            # implicitly cause an item call.  The problem is this is pretty
+            # unsound: there's no reason substituting an int with a Tensor is
+            # going to give the same results.  Today, you mostly get around
+            # this by typically not having capture_scalar_outputs on and graph
+            # breaking when someone tries to use the unspec variable in an
+            # int-y context.  But allowing it through here would break that.
+            # So don't.
+            #
+            # Once random values are setup to be represented as
+            # SymNodeVariable, this condition can be removed.  To check if
+            # you've done it right, this is a good test:
+            #
+            #   PYTORCH_TEST_WITH_DYNAMO=1 python test/test_reductions.py -k
+            #   TestReductionsCPU.test_dim_reduction_fns_fn_name_amax_cpu_bfloat16
+            and not isinstance(source, RandomValueSource)
+            and shape_env is not None
```

Comment:
when is shape_env None again ? presumably we'll still want to be able to handle float inputs even with static shapes

Reply:
In Dynamo, it is never none (even with static shapes), so we would be able to handle it there. However, FakeTensorMode can be used by hand and in some cases ShapeEnv is not allocated.